lib/record.gd

```diff
@@ -128,20 +128,32 @@ DeclareOperationKernel( "Unbind.", [ IsObject, IsObject ], UNB_REC );
 ##  <Attr Name="RecNames" Arg='record'/>
 ##
 ##  <Description>
-##  returns a list of strings corresponding to the names of the record
-##  components of the record <A>record</A>.
+##  returns a duplicate-free list of strings corresponding to the names of
+##  the record components of the record <A>record</A>.
+##  <P/>
+##  The list is sorted by <Ref Func="RNamObj" Label="for a string"/>
+##  for reasons of efficiency; see <Ref Oper="SortBy"/>.
+##  Therefore the ordering is consistent with one &GAP; session,
+##  but it is not necessarily consistent across different sessions.
+##  One could apply, for instance, <Ref Oper="Set"/> to obtain a result that
+##  is consistent across &GAP; sessions.
+##  <P/>
+##  Note that you cannot use the string result in the ordinary way to access
+##  or change a record component.
```

Comment:
This is not my text, as you can see I have just moved it from further down the manual entry, that’s why it appears in the diff.

Reply:
So in that sense I can’t answer for these sentences. However, my guess is that this text is trying to point out there’s a possible ambiguity about whether, when `var:=“name”` and accessing a record component, does `r.var` refers to the component of the record `r` with name `var`, or to the component of the record `r` with name `name`?

And it’s saying that GAP chooses to do the former. But the doc shows how to make the latter work. Because perhaps it’s a typical use cases that the values of record components are strings that define the names of other record components?

Again, just a guess.

Reply:
Ah yes, sorry, I did not notice that you just moved it, sorry. Then feel free to ignore it (although of course improving it would be nice, but it'd be unfair to block your PR over this!).

I agree with the direction of your guess.

Reply:
Once I’m back at my computer I’ll see if I can do anything to clarify this text, or perhaps whether I can justify removing it. Otherwise I’ll move it back to where it was, to remove it from the diff of this PR.